File: tests/nodejs/30_indirect_exec.js
/**
 * Case 30: JS 间接调用 - 变量函数调用
 * const f = require('child_process').exec; f(cmd)
 * 预期: 检出 CVI-3003 (eval/setTimeout) 或相关
 */

const cp = require('child_process');

function handleRequest(userInput) {
    // 间接调用: f 指向 eval, 参数是用户输入
    const f = eval;
    f(userInput);
}

File: tests/python/30_indirect_exec.py
#!/usr/bin/env python3
"""
Case 30: Python 间接调用 - globals()/getattr 动态调用
globals()['os.system'](cmd) 或 getattr(os, 'system')(cmd)
预期: 检出 CVI-7000 (命令执行)
"""
import os

def handle_request(user_input):
    # 通过 globals() 间接调用 os.system
    func = globals().get('os.system')
    if func:
        func(user_input)
